mantle/cmd/ore/aliyun/make-public.go

```diff
@@ -0,0 +1,86 @@
+// Copyright 2021 Red Hat
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package aliyun
+
+import (
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/spf13/cobra"
+)
+
+var (
+	cmdPublic = &cobra.Command{
+		Use:   "make-public <image:region...>",
+		Short: "Mark images on aliyun publicly available",
+		Long:  `Mark one or more images on aliyun publicly available`,
+		RunE:  makePublic,
+
+		SilenceUsage: true,
+	}
+)
+
+func init() {
+	Aliyun.AddCommand(cmdPublic)
+}
+
+func makePublic(cmd *cobra.Command, args []string) error {
+	if len(args) == 0 {
+		fmt.Fprintf(os.Stderr, "Specify one ore more image:region pairs.\n")
+		os.Exit(2)
+	}
+
+	for _, pair := range args {
+		if strings.Contains(pair, ":") == false {
+			fmt.Fprintf(os.Stderr, "Argument isn't a valid image:region pair: %v\n", pair)
+			os.Exit(2)
+		}
+
+		v := strings.Split(pair, ":")
+		if len(v) > 2 {
+			fmt.Fprintf(os.Stderr, "Argument isn't a valid image:region pair: %v\n", pair)
+			os.Exit(2)
+		}
+
+		image, region := v[0], v[1]
+		supportedRegions, err := API.ListRegions()
+		if err != nil {
+			fmt.Fprintf(os.Stderr, "could not list regions: %v\n", err)
+			os.Exit(2)
+		}
+
+		var validRegion = false
+		for _, r := range supportedRegions {
+			if region == r {
+				validRegion = true
+				break
+			}
+		}
+
+		if validRegion == false {
+			fmt.Fprintf(os.Stderr, "%v is not a valid region\n", region)
+			os.Exit(2)
+		}
```

Comment:
Is this actually needed? Won't `MakePublic` fail anyway if the region is invalid? Or is the idea to give a nicer error message?

Reply:
I'm trying to do some input validation before we start making calls to the Aliyun API